### ISSUE TITLE
Error when unfollowing (Unfollow loop error) on Raspberry Pi3b

### ISSUE THREAD

**Issue by pezzotto** [InstaPy/InstaPy#4616]
I'm having some troubles with my unfollow script.

I'm using instapy on raspian installed as the how-to. I can't understand why it's giving me this error after the first unfollow:

> ERROR [2019-07-04 09:50:38] [user]  Unfollow loop error:  'unfollow'
> 
> ERROR [2019-07-04 09:50:38] [user]  Unfollow loop error:  local variable 'unfollow_state' referenced before assignment
> 
> INFO [2019-07-04 09:50:38] [user]  --> Total people unfollowed : 0

Already updated instapy and deleted cookies, any idea?

## InstaPy configuration

```
# imports
from instapy import InstaPy
from instapy import smart_run

# login credentials
insta_username = 'xxxxxxxxxxxxx'
insta_password = 'xxxxxxxxxxxxx'

# get an InstaPy session!
# set headless_browser=True to run InstaPy in the background
session = InstaPy(username=insta_username,
                  password=insta_password,
                  headless_browser=False)
                  
with smart_run(session):
    """ Activity flow """
    # general settings
    
    session.set_quota_supervisor(enabled=True, sleep_after=["unfollows"], sleepyhead=True, stochastic_flow=True, notify_me=True,
                     peak_unfollows=(35, 402))
    
    """ First step of Unfollow action - Unfollow not follower users...
    """
    session.unfollow_users(amount=500, InstapyFollowed=(True, "nonfollowers"),
                           style="FIFO",
                           unfollow_after=18 * 60 * 60, sleep_delay=601)

    """ Second step of Massive Follow...
    """
    """session.follow_user_followers(['user1', 'user2', 'user3'], amount=800,
                                  randomize=False, interact=False)"""

    """ Second step of Unfollow action - Unfollow not follower users...
    
    session.unfollow_users(amount=500, InstapyFollowed=(True, "nonfollowers"),
                           style="FIFO",
                           unfollow_after=12 * 60 * 60, sleep_delay=601)"""

    """ Clean all followed user - Unfollow all users followed by InstaPy...
    """
    session.unfollow_users(amount=500, InstapyFollowed=(True, "all"),
                           style="FIFO", unfollow_after=36 * 60 * 60,
                           sleep_delay=601)

    """ Joining Engagement Pods...
    """
    session.join_pods()
```

**Comment by Luisetty** [comment]
I'm having the same issues and getting this on the log.

INFO [2019-07-04 10:06:40] [user]  --> Unfollowed 'user1'!
INFO [2019-07-04 10:06:40] [user]  	Removed '2019-06-20 20:37 ~ user1 ~ 6115834408' from user_followedPool.csv file

ERROR [2019-07-04 10:06:40] [user]  Unfollow loop error:  'unfollow'

ERROR [2019-07-04 10:06:40] [user]  Unfollow loop error:  local variable 'unfollow_state' referenced before assignment

INFO [2019-07-04 10:06:40] [user]  --> Total people unfollowed : 0

--

So it's not removing or removes 1 and says that didn't unfollow or i don't have any idea, I can post my file.py but I didn't want to open more threads but too boost this issue.

**Comment by Luisetty** [comment]
@pezzotto I found a quick fix and was to downgrade and use an older version.

it's working for me the 0.5.0, go to the pull versions and you can check the previous full releases. Probably they will fix it soon.

**Comment by soldiersebas** [comment]
EL mismo Problema ..Solucion??

**Comment by sydney-d** [comment]
Same issue here too.

**Comment by louisboulagnon** [comment]
Hey, the problem is linked to  #4620.

TL;DR : after the smart run, add `session.set_action_delays(enabled=True, like=2, comment=2, follow=3, unfollow=10)`

What's happening is that instapy looks for the unfollow sleep delay after unfollowing.
Because of the issue quoted above (bad parenthesis in `get_action_delays`), this try block will break before setting the unfollow and msg variables
https://github.com/timgrossmann/InstaPy/blob/44482c77524f295d5bd0b0220a6239705066ca6d/instapy/unfollow_util.py#L450-L462
And this line will throw the error (or the ones after, checking the msg variable), breaking the unfollow loop
https://github.com/timgrossmann/InstaPy/blob/44482c77524f295d5bd0b0220a6239705066ca6d/instapy/unfollow_util.py#L466
The quick fix before they fix the get_action_delays function would be to manually set the action delays. You can choose custom values if yout want (the bug 'cancels itself out' there and custom values work) but you have to fill all of them (otherwise other bugs will arise, e.g. #4619). the default are 
`defaults = {"like": 2, "comment": 2, "follow": 3, "unfollow": 10}`

**Comment by pezzotto** [comment]
@louisboulagnon thank you. Seems fixed now.

**Comment by IKDH** [comment]
Are you guys all able to do follow without being blocked by Instagram ?

**Comment by soldiersebas** [comment]
> ¿Todos ustedes pueden seguir sin ser bloqueados por Instagram?

esta pasando mucho, todas mis cuents han sido bloqueadas, aunque ya las desbbloquearon, todavía nadie ha dicho cuantos se pueden seguir por dia,   ALGUNO SABE PARA NO SER BLOQUEADO
